sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1761,4 +1763,67 @@ private static void checkEmptyStringAllowed(Class<?> type, JavaType genericType,
     }
     return supportedRunners.build();
   }
+
+  /**
+   * @param properties the properties to use to create the pipeline options instance.
+   * @return a pipeline options instance based on the specified properties.
+   */
+  public static PipelineOptions fromProperties(final Properties properties) {
+    if (properties == null) {
+      return create();
+    }
+    final String[] args = properties.stringPropertyNames()
+      .stream()
+      .map(k -> String.format("--%s=%s", k, properties.getProperty(k)))
+      .toArray(String[]::new);
+    return fromArgs(args).create();
+  }
+
+  /**
+   * @param prefix the prefix filter applied on system properties keys.
+   * @param properties the properties to use to create the pipeline options instance.
+   * @return a pipeline options instance based on the specified properties
+   * filtered with the specified prefix.
+   */
+  public static PipelineOptions fromProperties(final String prefix, final Properties properties) {
+    if (prefix == null) {
+      return fromProperties(properties);
+    }
+    if (properties == null) {
+      return create();
+    }
+    final Properties instantiationProperties = properties.stringPropertyNames()
+      .stream()
+      .filter(k -> k.startsWith(prefix))
+      .collect(
+        Properties::new,
```

Comment:
Is there any reason to collect this into a `Properties` object instead of just a `Map` or even just `List` of entries?

Reply:
overall idea is to have a single base signature. Choice is between Properties, Map<String, String>, List<Pair<String, String>> (or entry). I picked Properties because it is easy for system properties and to read from any source (properties.load) without any external dependency. Since it is a volatile data used in a single thread there is no concurrency issue so I choose to convert the only map manipulated by this code to a properties instance to conform to the other signatures.

Reply:
I guess my thought is that you should convert to a more basic type ASAP since properties are not relevant except for as an input format (just like command line is not relevant except as an input format). In this case, Properties is also basically just a Map anyhow so it doesn't bother me so much.

The fact that it is Map<Object, Object> may either be a bug or a feature; I'm not sure yet.

Reply:
Properties is an niput format yes, it is aligned with fromArgs and factories here which convert an inputformat to an options instance. Guess we can split it in N factories bit maybe overkill here. The object typing is inherited from a pre generics time but it should really be string/string and this is xhy i use the string keying which already filters it properly.

Map of strings can be used but is way less natural in end user code.

Same as previous comment: this is for users, not for beam dev so trying to embrace their future usage rather our own satisfaction.